HIP/hip-874.md

```diff
@@ -0,0 +1,53 @@
+requested-by: TierBot
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/pull/883
+---
+
+## Abstract:
+This HIP proposes an enhancement to the Hedera REST API to enable access to topic ID metadata, including the memo field and other relevant entity information. Currently, there is limited direct access to topic metadata through the REST API, which can hinder the development of applications that rely on this data for functionality or auditing purposes. This proposal aims to streamline access to topic metadata, reducing the need for workarounds and enhancing the capabilities of applications built on Hedera.
+
+## Motivation:
+The ability to access topic ID metadata, including the memo field, submit key, and deleted field directly through the REST API is crucial for a wide range of applications, including messaging, event tracking, and audit-ability functions. Currently, developers may need to employ indirect methods or multiple API calls to retrieve this information, leading to inefficiencies and potential inaccuracies. By providing direct access to this metadata, the proposal seeks to enhance developer experience, reduce computational load, and foster the development of more feature-rich applications on the Hedera platform.
+
+## Specification:
+The proposed enhancement involves extending the REST API to include a new endpoint or modify existing endpoints to return topic ID metadata, including the memo field and other relevant entity information.
+
+### Route:
+/api/v1/topics/{topic id}
+
+Current Response Structure returns:
+```json
+{"_status":{"messages":[{"message":"Not found"}]}}
+```
+
+Proposed Response Structure:
+```json
+{
+  "id": "0.0.123456",
+  "memo": "[Topic Memo]",
+  "admin_key": null,
+  "deleted": "[Boolean]",
+  "submit_key": null,
+  "auto_renew_period": 7890000,
+  "auto_renew_account_id": "0.0.654321",
+  "created_timestamp": "[Creation Time]"
+}
+```
+
+The addition of memo, created_timestamp, last_updated_time,  deleted, admin_key , auto_renew_period, submit_key ,auto_renew_account_id,id fields will provide comprehensive metadata about topics directly through the API.
+
+### Rationale:
+Providing direct access to topic ID metadata, including the memo and other relevant information, addresses a critical need for developers building on the Hedera network. This enhancement will facilitate a range of applications, from simple messaging systems to complex, multi-layered applications requiring detailed audit trails. The proposal supports the principles of transparency, efficiency, and developer friendliness that are essential to the Hedera ecosystem.
+
+### Backwards Compatibility:
+This change is designed to be backwards compatible, adding new fields to the API response without altering existing fields or endpoint behaviors. This ensures that existing applications will continue to operate without modification, while new and updated applications can take advantage of the enhanced metadata accessibility.
```

Comment:
It doesn't add new fields to the API response. It adds a whole new API, which is of course backwards compatible.

Reply:
Yup fair enough haha. Updated the language here.